nwb_conversion_tools/tools/roiextractors.py

```diff
@@ -0,0 +1,553 @@
+                        name="Accepted",
+                        description="1 if ROi was accepted or 0 if rejected as a cell during segmentation operation",
+                    ),
+                    VectorData(
+                        data=rejected_ids,
+                        name="Rejected",
+                        description="1 if ROi was rejected or 0 if accepted as a cell during segmentation operation",
+                    ),
+                ],
+                id=roi_ids,
+            )
+
+            ps = image_segmentation.create_plane_segmentation(**input_kwargs)
+
+        # Fluorescence Traces:
+        if "Flourescence" not in ophys.data_interfaces:
+            fluorescence = Fluorescence()
+            ophys.add(fluorescence)
+        else:
+            fluorescence = ophys.data_interfaces["Fluorescence"]
+        roi_response_dict = segext_obj.get_traces_dict()
+        roi_table_region = ps.create_roi_table_region(
+            description=f"region for Imaging plane{plane_no_loop}",
+            region=list(range(segext_obj.get_num_rois())),
+        )
+        rate = np.float("NaN") if segext_obj.get_sampling_frequency() is None else segext_obj.get_sampling_frequency()
+        for i, j in roi_response_dict.items():
+            data = getattr(segext_obj, f"_roi_response_{i}")
+            if data is not None:
+                data = np.asarray(data)
+                trace_name = "RoiResponseSeries" if i == "raw" else i.capitalize()
+                trace_name = trace_name if plane_no_loop == 0 else trace_name + f"_Plane{plane_no_loop}"
+                input_kwargs = dict(
+                    name=trace_name,
+                    data=data.T,
+                    rois=roi_table_region,
+                    rate=rate,
+                    unit="n.a.",
+                )
+                if trace_name not in fluorescence.roi_response_series:
+                    fluorescence.create_roi_response_series(**input_kwargs)
+
+        # create Two Photon Series:
+        if "TwoPhotonSeries" not in nwbfile.acquisition:
+            warn("could not find TwoPhotonSeries, using ImagingExtractor to create an nwbfile")
+
+        # adding images:
+        images_dict = segext_obj.get_images_dict()
+        if any([image is not None for image in images_dict.values()]):
+            images_name = "SegmentationImages" if plane_no_loop == 0 else f"SegmentationImages_Plane{plane_no_loop}"
+            if images_name not in ophys.data_interfaces:
+                images = Images(images_name)
+                for img_name, img_no in images_dict.items():
+                    if img_no is not None:
+                        images.add_image(GrayscaleImage(name=img_name, data=img_no.T))
+                ophys.add(images)
+
+        # saving NWB file:
+        if write:
+            io.write(nwbfile)
+            io.close()
+            # test read
+            with NWBHDF5IO(str(save_path), "r") as io:
+                io.read()
```

Comment:
is this necessary?

Reply:
Not something I would know :D copy and pasted from https://github.com/catalystneuro/roiextractors/blob/master/roiextractors/extractors/nwbextractors/nwbextractors.py#L1019

I'm guessing not? Should probably be deleted; it's not standard practice for the writing functions to check successful post write read... Also not terribly relevant since the read doesn't attempt to access any data from the objects